File: staircase/staircase.go
package main
import (
    "fmt"
    "strings"
)

func main() {
    var n int
    fmt.Scanf("%v\n",&n)
    for i := 1; i <= n; i++ {
        str := ""
        hashes := i
        spaces := (n-i)
        str += strings.Repeat(" ", spaces)
        str += strings.Repeat("#", hashes)
        fmt.Println(str)
    }
}

File: staircase/staircase.js
function main() {
  var n = parseInt(readLine());
  for (var i = 1; i < n + 1; i++) {
    var string = ''
    var hashes = i;
    var spaces = n - i;
    string += ' '.repeat(spaces);
    string += '#'.repeat(hashes);
    console.log(string);
  }
}
